tests/test_block.py

```diff
@@ -22,10 +22,12 @@ def test_save(self):
                                   'type': 'type',
                                   'key': 'val'})
 
-        def test_save_instance_param(self):
+        def test_save_with_no_instance(self):
             b = Block('name', 'type')
             b.config = {'key': 'val'}
             b._put = MagicMock()
-            b.save('instance')
-            self.assertTrue(b._put.called)
-            self.assertEqual(b._instance, 'instance')
+            with self.assertRaises(Exception) as context:
+                b.save()
+            self.assertTrue('Block is not associated with an instance',
+                            context.exception)
```

Comment:
I think you have this backwards. Right now you are asserting that a string is True